tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -1966,6 +1937,327 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
+						}
+						for _, pvc := range pvcs.Items {
+							annotations := pvc.GetObjectMeta().GetAnnotations()
+							log.Logf("pvc annotations: %+v", annotations)
+							_, ok := annotations["tidb.pingcap.com/pvc-defer-deleting"]
+							framework.ExpectEqual(ok, false, "expect PVC %s/%s not to have annotation tidb.pingcap.com/pvc-defer-deleting", pvc.GetNamespace(), pvc.GetName())
+							pvcUIDString := pvcUIDs[pvc.Name]
+							framework.ExpectNotEqual(string(pvc.UID), pvcUIDString)
+						}
+					}
+				})
+			}
+		})
+
+		ginkgo.Context("while concurrently upgrade", func() {
+			components := []v1alpha1.MemberType{"pd", "tikv", "tidb"}
+			for _, comp := range components {
+				comp := comp
+				var replicasLarge, replicasSmall int32
+				switch comp {
+				case v1alpha1.PDMemberType:
+					replicasLarge = 5
+					replicasSmall = 3
+				case v1alpha1.TiKVMemberType:
+					replicasLarge = 4
+					replicasSmall = 3
+				case v1alpha1.TiDBMemberType:
+					replicasLarge = 3
+					replicasSmall = 2
+				}
+				ginkgo.It(fmt.Sprintf("should work for %s", comp), func() {
+					ginkgo.By("Deploy initial tc")
+					tc := fixture.GetTidbCluster(ns, fmt.Sprintf("scale-in-upgrade-%s", comp), utilimage.TiDBV4Prev)
+					switch comp {
+					case v1alpha1.PDMemberType:
+						tc.Spec.PD.Replicas = replicasLarge
+					case v1alpha1.TiKVMemberType:
+						tc.Spec.TiKV.Replicas = replicasLarge
+					case v1alpha1.TiDBMemberType:
+						tc.Spec.TiDB.Replicas = replicasLarge
+					}
+					utiltc.MustCreateTCWithComponentsReady(genericCli, oa, tc, 5*time.Minute, 10*time.Second)
+
+					ginkgo.By(fmt.Sprintf("Scale in %s to %d replicas", comp, replicasSmall))
+					err := controller.GuaranteedUpdate(genericCli, tc, func() error {
+						switch comp {
+						case v1alpha1.PDMemberType:
+							tc.Spec.PD.Replicas = replicasSmall
+						case v1alpha1.TiKVMemberType:
+							tc.Spec.TiKV.Replicas = replicasSmall
+						case v1alpha1.TiDBMemberType:
+							tc.Spec.TiDB.Replicas = replicasSmall
+						}
+						return nil
+					})
+					framework.ExpectNoError(err, "failed to scale in %s for TidbCluster %s/%s", comp, ns, tc.Name)
+
+					ginkgo.By(fmt.Sprintf("Wait for %s to be in ScalePhase", comp))
+					utiltc.MustWaitForComponentPhase(cli, tc, comp, v1alpha1.ScalePhase, 3*time.Minute, 10*time.Second)
+
+					ginkgo.By(fmt.Sprintf("Upgrade %s version concurrently", comp))
+					err = controller.GuaranteedUpdate(genericCli, tc, func() error {
+						switch comp {
+						case v1alpha1.PDMemberType:
+							tc.Spec.PD.Version = pointer.StringPtr(utilimage.TiDBV4)
+						case v1alpha1.TiKVMemberType:
+							tc.Spec.TiKV.Version = pointer.StringPtr(utilimage.TiDBV4)
+						case v1alpha1.TiDBMemberType:
+							tc.Spec.TiDB.Version = pointer.StringPtr(utilimage.TiDBV4)
+						}
+						return nil
+					})
+					framework.ExpectNoError(err, "failed to scale in %s for TidbCluster %s/%s", comp, ns, tc.Name)
```

Comment:
please fix this message and the one in L2154.

Reply:
What do you mean by fix?

Reply:
not `scale in` here, right?

Reply:
fixed